### PR TITLE
Splitter: fallback for setting incorrect separatorSize value

### PR DESCRIPTION
**Pull request by AlexanderMoiseev**

### CODE REVIEW

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 1]
```diff
@@ -478,7 +480,8 @@ class Splitter extends (CollectionWidget as any) {
     this._resizeHandles.forEach((resizeHandle) => {
       const { disabled, separatorSize } = resizeHandle.option();
 
-      size += disabled ? INACTIVE_RESIZE_HANDLE_SIZE : separatorSize as number;
+      size += disabled ? INACTIVE_RESIZE_HANDLE_SIZE
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: Let's move this logic to resizeHandle. We can create
```
getSize() { return this._isInActive() ? INACTIVE_RESIZE_HANDLE_SIZE
      : getSafeSize(separatorSize, DEFAULT_RESIZE_HANDLE_SIZE); }
``` 
method in resizeHandle and use its here and inside resizeHandle. In this case INACTIVE_RESIZE_HANDLE_SIZE can be removed from this file

File: packages/devextreme/js/__internal/ui/splitter/resize_handle.ts [hunk 2]
```diff
@@ -180,15 +182,16 @@ class ResizeHandle extends (Widget as any) {
...
-    if (resizable === false && showCollapseNext === false && showCollapsePrev === false) {
-      this.option('disabled', true);
-      this.option(dimension, INACTIVE_RESIZE_HANDLE_SIZE);
-      this.option(inverseDimension, null);
-    } else {
-      this.option(dimension, separatorSize);
-      this.option(inverseDimension, null);
-      this.option('disabled', false);
-    }
+    const disabled = resizable === false
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/resize_handle.ts]: Let's extract it into _isInactive() {}

Comment (EugeniyKiyashko) [reply packages/devextreme/js/__internal/ui/splitter/resize_handle.ts]: I think it's better to rename '_setResizeHandleSize' into '_updateDimensions'

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js [hunk 3]
```diff
@@ -1335,6 +1335,49 @@ QUnit.module('Behavoir', moduleConfig, () => {
...
+                this.assertLayout([50, 50]);
+
+                const $resizeHandle = this.getResizeHandles();
+
+                assert.strictEqual($resizeHandle.css(dimension), '8px');
+            });
+        });
+    });
+
+    [
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: ['horizontal', 'vertical'].forEach((orientation) => {
    ['', '', '',].forEach((separatorSize) => {
         QUnit.test(`Resize handle should fallback to default if separatorSize is incorrect on init, separatorSize=${sepratorSize}, orientation=${orientation}`) {
         }
    })
})

The dimension value is auto-calculated and depends on the orientation value

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js [hunk 4]
```diff
@@ -200,6 +214,17 @@ QUnit.module('ResizeHandle markup', moduleConfig, () => {
                 assert.strictEqual(this.$element.css(newDimension), '5px');
                 assert.notStrictEqual(this.$element.css(dimension), '5px');
             });
+
+            QUnit.test(`Resize handle ${dimension} should fallback to default if separatorSize changed at runtime to incorrect value (direction=${direction})`, function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js]: Move this test into 
```
['50vh',  '20spx', 'd10',  ...].forEach((separatorSize) => {
```

File: packages/devextreme/js/__internal/ui/splitter/resize_handle.ts [hunk 5]
```diff
@@ -474,6 +476,13 @@ class ResizeHandle extends (Widget as any)<ResizeHandleOptions> {
         super._optionChanged(args);
     }
   }
+
+  getSize(): number {
+    const { separatorSize } = this.option();
+
+    return this._isInactive() ? INACTIVE_RESIZE_HANDLE_SIZE
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/resize_handle.ts]: move ? ... to the next line

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 6]
```diff
@@ -475,10 +473,8 @@ class Splitter extends (CollectionWidget as any) {
   _getResizeHandlesSize(): number {
     let size = 0;
 
-    this._resizeHandles.forEach((resizeHandle) => {
-      const { disabled, separatorSize } = resizeHandle.option();
-
-      size += disabled ? INACTIVE_RESIZE_HANDLE_SIZE : separatorSize as number;
+    this._resizeHandles.forEach((resizeHandle: ResizeHandle) => {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: _getResizeHandlesSize(): number {
  return this._resizeHandles.reduce((totalSize, resizeHandle) => totalSize + resizeHandle.getSize(), 0);
}

File: packages/devextreme/js/__internal/ui/splitter/resize_handle.ts [hunk 7]
```diff
@@ -148,7 +150,7 @@ class ResizeHandle extends (Widget as any)<ResizeHandleOptions> {
     this.$element().addClass(RESIZE_HANDLE_CLASS);
     this.$element().toggleClass(RESIZE_HANDLE_RESIZABLE_CLASS, resizable);
     this._toggleDirectionClass();
-    this._setResizeHandleSize();
+    this._updateDimensions();
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/resize_handle.ts:152]: Please, to be sure that _updateDimensions doesn't call upon window resizing

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js [hunk 8]
```diff
@@ -168,6 +168,32 @@ QUnit.module('ResizeHandle markup', moduleConfig, () => {
                 assert.strictEqual(this.$element.css(dimension), '8px');
             });
 
+            [
+                { separatorSize: '50vh' }, { separatorSize: '20spx' }, { separatorSize: 'd10' }, { separatorSize: 'NaN' },
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js]: add value with spaces ' 100 ', ' 100px ',

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js [hunk 9]
```diff
@@ -1386,6 +1387,50 @@ QUnit.module('Behavoir', moduleConfig, () => {
         assert.strictEqual($resizeHandle.css('width'), '5px');
     });
 
+    [
+        { orientation: 'horizontal' },
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: ['horizontal', 'vertical'].forEach(() => {})

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js [hunk 10]
```diff
@@ -1386,6 +1387,50 @@ QUnit.module('Behavoir', moduleConfig, () => {
...
+                const $resizeHandle = this.getResizeHandles();
+                const dimension = orientation === 'horizontal' ? 'width' : 'height';
+
+                assert.strictEqual($resizeHandle.css(dimension), '8px');
+            });
+        });
+    });
+
+    [
+        { orientation: 'horizontal', dimension: 'width' },
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: ['horizontal', 'vertical'].forEach(() => {})

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js [hunk 11]
```diff
@@ -1386,6 +1387,50 @@ QUnit.module('Behavoir', moduleConfig, () => {
         assert.strictEqual($resizeHandle.css('width'), '5px');
     });
 
+    [
+        { orientation: 'horizontal' },
+        { orientation: 'vertical' },
+    ].forEach(({ orientation }) => {
+        [
+            { separatorSize: '50vh' }, { separatorSize: '20spx' }, { separatorSize: 'd10' }, { separatorSize: 'NaN' },
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: ['50vh', '20spx', ..].forEach(() => {})

File: packages/devextreme/js/__internal/ui/splitter/resize_handle.ts (hunk 7 above)
Comment (AlexanderMoiseev) [reply packages/devextreme/js/__internal/ui/splitter/resize_handle.ts:152]: It is not called on resizing the window - if thats what you mean

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js (hunk 8 above)
Comment (AlexanderMoiseev) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js:172]: why do you think ' 100' is a bad input?

File: packages/devextreme/js/__internal/ui/splitter/utils/layout.ts [hunk 12]
```diff
@@ -377,6 +377,10 @@ function adjustAndDistributeLayoutSize(
   return nextLayout;
 }
 
+export function getSafeSize(size: number, fallBackSize: number): number {
+  return typeof size === 'number' ? size : fallBackSize;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/utils/layout.ts]: Better to use Number.isFinite(value). I think that this suits for us requirenment

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 13]
```diff
@@ -473,15 +472,8 @@ class Splitter extends (CollectionWidget as any) {
...
-
-    this._resizeHandles.forEach((resizeHandle) => {
-      const { disabled, separatorSize } = resizeHandle.option();
-
-      size += disabled ? INACTIVE_RESIZE_HANDLE_SIZE : separatorSize as number;
-    });
-
-    return size;
+    // eslint-disable-next-line max-len
+    return this._resizeHandles.reduce((size: number, rh: ResizeHandle) => size + rh.getSize(), 0) as number;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: No need to shorten variable names; it's unreadable as it is now. rh -> resizeHandle. Add line-brakes to improve formatting.

Comment (EugeniyKiyashko) [reply packages/devextreme/js/__internal/ui/splitter/splitter.ts]: Why do you need to cast reduce resize?

File: packages/devextreme/js/__internal/ui/splitter/splitter.ts [hunk 14]
```diff
@@ -431,18 +431,17 @@ class Splitter extends (CollectionWidget as any) {
       },
       onResizeEnd: (e: ResizeEndEvent): void => {
         const { element, event } = e;
-        const $resizeHandle = $(element);
 
         this._feedbackDeferred.resolve();
-        this._toggleActiveState($resizeHandle, false);
+        this._toggleActiveState($(getPublicElement($(element))), false);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/splitter.ts]: Revert excess changes

File: packages/devextreme/js/__internal/ui/splitter/resize_handle.ts [hunk 15]
```diff
@@ -14,6 +14,7 @@ import {
   getActionNameByEventName,
   RESIZE_EVENT,
 } from './utils/event';
+import { getSafeSize } from './utils/layout';
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/splitter/resize_handle.ts]: Let's not create it in method. And better create correct isNumber() method in separate util file

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js [hunk 16]
```diff
@@ -168,6 +168,29 @@ QUnit.module('ResizeHandle markup', moduleConfig, () => {
                 assert.strictEqual(this.$element.css(dimension), '8px');
             });
 
+            ['50vh', '20spx', 'd10', 'NaN', '2%', '20em', '1vw', '', '100px ', '100px ', ' 11 ', '12'].forEach((separatorSize) => {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/resizeHandle.markup.tests.js]: Check on NaN and negative numbers

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js [hunk 17]
```diff
@@ -1386,6 +1387,40 @@ QUnit.module('Behavoir', moduleConfig, () => {
         assert.strictEqual($resizeHandle.css('width'), '5px');
     });
 
+    ['horizontal', 'vertical'].forEach((orientation) => {
+        ['50vh', '20spx', 'd10', 'NaN', '2%', '20em', '1vw', '', ' 100px', '100px ', ' 11 ', '12'].forEach((separatorSize) => {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: Check on NaN and negative number

Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/splitter.tests.js]: + undefined, null

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**